### PR TITLE
Using ConditionSet as unsolvable objects instead of NotImplementedError.

### LINKED ISSUE #9773: Use ConditionSet as an unsolved object for solveset
In https://github.com/sympy/sympy/pull/9696 we implemented a `ConditionSet` object which can be used to represent unsolved equations and inequalities as a Set object. We need to use it as an unsolved object in solveset.

### PR DESCRIPTION
**Pull request by kumarkrishna**
Fixes #9773.
TODO:
- [x] real equations
- [x] complex equations
- [x] inequalities
- [x] Edit Docstring.
- [x] Update tests.

### CODE REVIEW

**Comment by kumarkrishna** [comment]
For some weird reason this fixes #9759 . @aktech @hargup @gxyd

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -464,7 +464,7 @@ def solveset_real(f, symbol):
                 else:
                     result += solveset_real(equation, symbol)
         else:
-            raise NotImplementedError
+            result = ConditionSet(Lambda(symbol, f), S.Reals)
```
Comment (aktech) [inline sympy/solvers/solveset.py]: `f`  --> `Eq(f, 0)`
The second argument to `Lambda` should be a relational for `ConditionSet`, and `f` is an expression.

File: sympy/solvers/solveset.py [hunk 2]
```diff
@@ -497,7 +497,7 @@ def _solve_real_trig(f, symbol):
     g, h = g.expand(), h.expand()
     g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
     if g.has(symbol) or h.has(symbol):
-        raise NotImplementedError
+        return ConditionSet(Lambda(symbol, f), S.Reals)
```
Comment (aktech) [inline sympy/solvers/solveset.py]: `f`  --> `Eq(f, 0)`

File: sympy/solvers/solveset.py [hunk 3]
```diff
@@ -507,7 +507,7 @@ def _solve_real_trig(f, symbol):
     elif solns is S.EmptySet:
         return S.EmptySet
     else:
-        raise NotImplementedError
+        return ConditionSet(Lambda(symbol,f), S.Reals)
```
Comment (aktech) [inline sympy/solvers/solveset.py]: `f`  --> `Eq(f, 0)`

File: sympy/solvers/solveset.py [hunk 4]
```diff
@@ -886,9 +882,7 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if f.is_Relational:
         if not domain.is_subset(S.Reals):
-            raise NotImplementedError("Inequalities in the complex domain are "
-                                      "not supported. Try the real domain by"
-                                      "setting domain=S.Reals")
+            return ConditionSet(Lambda(symbol, f), S.Complexes)
```
Comment (aktech) [inline sympy/solvers/solveset.py]: I think this should not be returned unevaluated. Users should know they can't solve complex inequalities from `solveset`.

**Comment by gxyd** [comment]
returning a `ConditionalSet` instead of the `NotImplementedError`, I am not sure if we want the solution to be returned in the form of an `un-evaluated set`. For cases where currently the algorithm is not there to find the solution, it is still to be replaced by some algorithm which could solve the problem. For your `PR` Returning `ConditionalSet` also raises error for `containment` of elements.
Probably mostly we want an `evaluated set` not `un-evaluated set`.

**Comment by hargup** [comment]
Unevaluated objects are a convention in SymPy and I think the reason is that often unevaluated objects have methods which are atleast more useful than raising an error. For example `RootOf` object has `evalf` attribute which gives the floating point approximation of the root. In our case `ConditionSet` has the `contains` method. A bigger advantage of `ConditionSet` is that it allows us to return partial solutions, for example after this PR is complete we will have:
```python
>>> solveset(x*(x - sin(x)), x, domain=S.Reals)
{0}UConditionSet(Lambda(x, Eq(-x + sin(x), 0)), (-oo, oo))
```

**Comment by kumarkrishna** [comment]
@hargup , the `ConditionSet` intersection takes care of the membership of the partial solutions as well, improving the represented solution. For example, 
```
>>> x = Symbol('x', real=True)
>>> solveset(x*(x - sin(x)), x, domain=S.Reals)
ConditionSet(Lambda(x, Eq(x - sin(x), 0)), (-oo, oo))
```
@gxyd The current PR should also fix errors for `containment` of elements now, I believe.
```
>>> 0 in solveset(x*(x - sin(x)), x, domain=S.Reals)
True
>>> 1 in solveset(x*(x - sin(x)), x, domain=S.Reals)
False
```

**Comment by hargup** [comment]
@kumarkrishna Thanks for your work, this PR looks good. Can you add more tests and give a coverage report so that we can make sure all the lines which return `ConditionSet`s are covered. The coverage report for the solvers module can be generated by `./bin/coverage sympy/solvers/`. Also are you returning `ConditionSet` for inequalities, if not please add it.

**Comment by kumarkrishna** [comment]
The current version of the PR supports unsolved inequalities. As for the coverage report, 6/12 instances of `ConditionsSet` are covered. @hargup

**Comment by kumarkrishna** [comment]
@hargup The `pass` statement was not required, updated changes.

**Comment by hargup** [comment]
+1, will merge when the test passes.

File: sympy/solvers/tests/test_solveset.py [hunk 5]
```diff
@@ -808,6 +811,29 @@ def test_solveset():
                                                   S.Integers)
 
 
+def test_conditonset():
```
Comment (flacjacket) [inline sympy/solvers/tests/test_solveset.py]: Not sure what happened between this PR passing the tests and this being merged, but this test is now failing in master.

Comment (aktech) [inline sympy/solvers/tests/test_solveset.py]: That's because #9591  was Merged in between.

**Pull request merged**